dashboard/lib/services/lesson_import_helper.rb

```diff
@@ -0,0 +1,279 @@
+    [create_activity_with_levels(script_levels, lesson_id, 1)]
+  end
+
+  def self.create_activity_sections(activity_markdown)
+    # Find any special syntax, such as tips or remarks, and gather them.
+    # TODO tips and remarks both show up in tip_matches. We filter out remarks
+    # but we should try to do something a bit smarter here.
+    tip_matches = find_tips(activity_markdown).select {|m| m[1] != 'say'}.map {|m| {index: activity_markdown.index(m[0]), type: 'tip', match: m, substring: m[0]}}
+    tip_link_matches = find_tip_links(activity_markdown).map {|m| {index: activity_markdown.index(m[0]), type: 'tiplink', match: m, substring: m[0]}}
+    remark_matches = find_remarks(activity_markdown).map {|m| {index: activity_markdown.index(m[0]), type: 'remark', match: m, substring: m[0]}}
+    skippable_matches = find_skippable_syntax(activity_markdown).map {|m| {index: activity_markdown.index(m[0]), type: 'skippable', match: m, substring: m[0]}}
+    slide_matches = find_slides(activity_markdown).map {|m| {index: activity_markdown.index(m[0]), type: 'slide', match: m, substring: m[0]}}
+    matches = tip_matches + remark_matches + tip_link_matches + skippable_matches + slide_matches
+    sorted_matches = matches.sort_by {|m| m[:index]}
+    return [ActivitySection.new(description: activity_markdown.strip, key: SecureRandom.uuid, position: 1)] if matches.empty?
+
+    sorted_matches = find_markdown_chunks(activity_markdown, sorted_matches)
+
+    # Create a map of tip key -> an array of tip matches
+    # Sometimes, a key is used multiple times in the same activity. To handle
+    # this we'll, just pair them with the tip links in order.
+    tip_match_map = Hash.new([])
+    tip_matches.each do |match|
+      key = match[:match][3]&.strip || "#{match[:match][1]}-0"
+      tip_match_map[key].push(match)
+    end
+
+    sections = []
+    slide = false
+    position = 1
+    sorted_matches.each do |match|
+      activity_section = nil
+      if match[:type] == 'skippable' || match[:type] == 'tip'
+        next
+      elsif match[:type] == 'slide'
+        # For now, if we find the slide token, apply it to the next match
+        slide = true
+        next
+      elsif match[:type] == 'tiplink'
+        activity_section = create_activity_section_with_tip(match[:match], tip_match_map)
+      elsif match[:type] == 'remark'
+        activity_section = create_activity_section_with_remark(match[:match])
+      else
+        activity_section = create_basic_activity_section(match[:substring].strip)
+      end
+      next unless activity_section
+      activity_section.position = position
+      position += 1
+      activity_section.slide = slide
+      slide = false
+      activity_section.key ||= SecureRandom.uuid
+      sections = sections.push(activity_section)
+    end
+
+    # If there are any tips that didn't have a match, put them at the end
+    # TODO ideally, we'd put them in the spot they were written.
```

Comment:
It would be interesting for this to print out what lessons this is happening on so we can investigate it

Reply:
I'm adding a puts statement -- it looks like this happens in a fair number of lessons. Some might be bugs in the script to be fair but not all (see [this lesson](https://curriculum.code.org/csf-20/coursea/2/))